machine-runner/src/utils/type-utils.ts

```diff
@@ -4,6 +4,8 @@ export type DeepReadonly<T> = {
 
 export type NonZeroTuple<T> = [T, ...T[]]
 
+export type RetvalOrElse<T, Else> = T extends (...args: any[]) => infer Retval ? Retval : Else
```

Comment:
Does this read "if we are able to infer `Retval`, use it, otherwise use `Else`"?

Reply:
Exactly!